src/verifai/features/features.py

```diff
@@ -1100,33 +1164,61 @@ def meaningOfFlatCoordinate(self, index, pointName='point'):
-        for name, feature in self.namedFeatures:
+        for name, feature in self.staticFeatureNamed.items():
             domain = feature.domain
             if feature.lengthDomain:
                 if index == 0:
-                    return f'len({pointName}.{name})'
+                    return f'len({pointName}.staticSample.{name})'
                 else:
                     index -= 1
                     elem = index // domain.flattenedDimension
                     if elem < feature.maxLength:
                         subIndex = index % domain.flattenedDimension
-                        subPoint = f'{pointName}.{name}[{elem}]'
+                        subPoint = f'{pointName}.staticSample.{name}[{elem}]'
                         return domain.meaningOfFlatCoordinate(subIndex,
                             pointName=subPoint)
                     index -= feature.maxLength * domain.flattenedDimension
             else:
                 if index < domain.flattenedDimension:
-                    subPoint = f'{pointName}.{name}'
+                    subPoint = f'{pointName}.staticSample.{name}'
                     return domain.meaningOfFlatCoordinate(index,
                                                           pointName=subPoint)
                 index -= domain.flattenedDimension
+
+        if index == 0:
+            return f'len({pointName}.dynamicSampleHistory)'
+        index -= 1
+
+        for name, feature in self.dynamicFeatureNamed.items():
+            domain = feature.domain
+            for time_i in range(self.timeBound):
+                if feature.lengthDomain:
+                    if index == 0:
+                        return f'len({pointName}.dynamicSampleHistory[{time_i}].{name})'
+                    else:
+                        index -= 1
+                        elem = index // domain.flattenedDimension
+                        if elem < feature.maxLength:
+                            subIndex = index % domain.flattenedDimension
+                            subPoint = f'{pointName}.dynamicSampleHistory[{time_i}].{name}[{elem}]'
+                            return domain.meaningOfFlatCoordinate(subIndex,
+                                pointName=subPoint)
+                        index -= feature.maxLength * domain.flattenedDimension
+                else:
+                    if index < domain.flattenedDimension:
+                        subPoint = f'{pointName}.dynamicSampleHistory[{time_i}].{name}'
+                        return domain.meaningOfFlatCoordinate(index,
+                                                            pointName=subPoint)
+                    index -= domain.flattenedDimension
+
         raise RuntimeError('impossible index arithmetic')
 
     def pandasIndexForFlatCoordinate(self, index):
         """Pandas index of a coordinate of a flattened point in this space.
 
         See meaningOfFlatCoordinate, and Domain.pandasIndexForFlatCoordinate.
         """
+        #TODO Update
```

Comment:
Honestly I can't remember what this method is supposed to be for: I can't find any usages of it in my code at all. If updating it to support time series is complex, we could remove it. But since it is documented, we should not leave it in a broken state.